versioned/persist/adapter/src/main/java/org/projectnessie/versioned/persist/adapter/spi/FetchValuesUsingOpenAddressing.java

```diff
@@ -58,28 +60,25 @@ final class FetchValuesUsingOpenAddressing {
 
     entityIdToSegment = new Object2IntHashMap<>(keyListCount, Hashing.DEFAULT_LOAD_FACTOR, -1);
 
-    segmentMask = entry.getKeyListSegmentCount() - 1;
+    openAddressingMask = entry.getKeyListBucketCount() - 1;
 
     keyListIds = entry.getKeyListsIds();
   }
 
   /** Calculates the open-addressing bucket for a key. */
-  int segmentForKey(Key key) {
-    return segment(key.hashCode());
-  }
-
-  private int segment(int num) {
-    return num & segmentMask;
+  int bucketForKey(Key key) {
+    return key.hashCode() & openAddressingMask;
   }
 
   /**
    * Identifies the segment for a bucket. Segment 0 is the embedded key list, segment 1 is the first
    * key-list-entity.
    */
+  @VisibleForTesting
   int segmentForKey(int bucket, int round) {
     int binIdx = Arrays.binarySearch(keyListSegmentOffsets, bucket);
     int segment = binIdx >= 0 ? binIdx + 1 : -binIdx - 1;
-    return segment(segment) + round;
+    return (segment + round) % keyListsArray.length;
```

Comment:
I'm very curious about this line and whether you think this looks valid, since I know it just went through an edit in #4867.  However, the old behavior of ANDing the segment index against the bucket index mask seemed like a no-op under sane configurations, and could actually trigger an erroneous miss in a pathologically bad configuration (exercised by one of the new ITs).

Reply:
Yea, you're right here - also your assumption "b/c of bad nomenclature" is also right.
`(segment + round) % keyListsArray.length` looks correct to me now - and this mod-operation is probably the reason why it's incorrectly been a mod-operation in the above methods above.
TL;DR very good catch!